src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -101,7 +105,7 @@ internal sealed class DiagnosticData : IEquatable<DiagnosticData>
         /// some diagnostic data such as created from build will have original line/column but not text span
         /// in those cases, use GetTextSpan method instead to calculate one from original line/column
         /// </summary>
-        public TextSpan TextSpan => (DataLocation?.SourceSpan).Value;
+        public TextSpan TextSpan => (DataLocation?.SourceSpan)!.Value;
```

Comment:
💡 This would benefit from explicitly throwing the exception:

```suggestion
        public TextSpan TextSpan => DataLocation?.SourceSpan ?? throw new InvalidOperationException(...);
```